src/app/shared/directives/hide-if-not-authenticated.directive.ts

```diff
@@ -0,0 +1,90 @@
+import { ChangeDetectorRef, Directive, HostListener, Input, OnChanges, OnDestroy, OnInit, SimpleChanges, TemplateRef, ViewContainerRef } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+import { AuthenticationService } from '../services/authentication.service';
+
+
+@Directive({
+  selector: '[appHideIfNotAuthenticated]'
+})
+export class HideIfNotAuthenticatedDirective implements OnInit, OnChanges {
+
+  constructor(
+    private templateRef: TemplateRef<unknown>,
+    private viewContainer: ViewContainerRef,
+    private authService: AuthenticationService
+  ) {}
+
+
+  @HostListener('click')
+  onClick(): void {
+    this.updateRendering();
+  }
+
+  ngOnInit(): void {
+    this.updateRendering();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    this.updateRendering();
+  }
+
+  private updateRendering(): void {
+    const shouldHide: boolean = !this.authService.isAuthenticated();
+
+    if (shouldHide) {
+      this.viewContainer.clear();
+      console.log('should not be rendered in custom directive');
+    } else {
+      console.log('should render in custom directive');
+      this.viewContainer.createEmbeddedView(this.templateRef);
+    }
+  }
+
+
+  // ngOnInit(): void {
```

Comment:
Please delete the commented code - the directive should seem to work like intended :)
Well done!

Reply:
Sure, just forgot)